javascript/javascript1/week3/smart-phone-usage-app.js

```diff
@@ -0,0 +1,101 @@
+// CactusIO-interactive (Smart phone usage app) optional
+
+// Adding an activity
+const activities = [];
+let today = new Date().toLocaleDateString("da-DK"); //localization is not supported in node
+
+function addActivity(activity, duration) {
+    if (typeof (activity) === "string" && typeof (duration) === "number") {
+        activities.push({
+            acitivityDate: today,
+            //acitivityDate: '17.09.2019', only for testing showStatusToday function
+            activityName: activity,
+            activityDuration: duration
+        });
+    }
+}
+addActivity("Youtube", 120);
+addActivity("Facebook", 30);
+addActivity("Google", 200);
+
+console.log(activities);
+
+// Show my status
+
+function showStatus() {
+    let sumDuration = 0;
+
+    const limit = 300;
+
+    if (activities.length === 0) {
+        console.log(`Add some activities before calling showStatus`);
+    } else {
+
+        for (let i = 0; i < activities.length; i++) {
+            sumDuration += activities[i].activityDuration;
+        }
+        console.log(`You have added ${activities.length} activities. They amount to ${sumDuration} min of usage.`);
+
+        if (sumDuration >= limit) {
+            console.log(`You have reached your limit, no more smartphoning for you!`);
+        }
+    }
+}
+
+
+showStatus();
+
+
+// Improve the showStatus function by only showing the number of actitivies for that specific day.
+
+function showStatusToday() {
+    let sumDuration = 0;
+
+
+    if (activities.length === 0) {
+        console.log(`Add some activities before calling showStatus`);
+    } else {
+        const todayActivities = [];
+
+        for (let i = 0; i < activities.length; i++) {
+            if (activities[i].acitivityDate === today) {
+
+                sumDuration += activities[i].activityDuration;
+                todayActivities.push(activities[i]);
+
+            }
+        }
+
+        // Extra feature
+        // daily percentage 
+        const minutesInDay = 24 * 60;
+        if (todayActivities.length !== 0) {
+            console.log(`You have added ${todayActivities.length} activities today. They amount to ${sumDuration} min of usage. That is ${(( sumDuration * 100 ) / minutesInDay).toFixed(2) }% of daily time.`);
+        }
+
+
+    }
+}
+showStatusToday();
+
+
+
+// Create a function for calculating the activity a user has spent the most time on.
+
+function mostTimeOn() {
+    if (activities.length !== 0) {
```

Comment:
Generally speaking there's no need to check if activities is empty array
loops just won't do anything if array is empty. 
This simplifies code a little bit